store.go

```diff
@@ -2427,6 +2503,18 @@ func (s *store) ContainerRunDirectory(id string) (string, error) {
 		return "", err
 	}
 
+	rootUID, rootGID, err := idtools.GetRootUIDGID(s.uidMap, s.gidMap)
```

Comment:
Will this mean that the user data ends up being written to /run?   Won't this cause issues on reboot?  I think this data needs to be bind mounted from the host onto /run, just like /var/lib/containers/storage/overlay

Reply:
ah true, I need to take care of that.

I've pushed another version that implements this part.  I will test it better tomorrow though